cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,323 @@
+                        }
+                        break;
+                    default:
+                        showMessage = false;
+                        output("Invalid operation, please select one of [OperationType/SequenceNumber/SegmentId/Offset/Length/Attributes]");
+                }
+                predicate = clause.equals("and") ?
+                        predicates.stream().reduce(Predicate::and).orElse( x -> true) : predicates.stream().reduce(Predicate::or).orElse( x -> false);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                outputException(ex);
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                outputException(ex);
+            }
+            if (showMessage) {
+                output("You can continue adding conditions for inspect.");
+            }
+            finishInputCommands = !confirmContinue();
+            next = !finishInputCommands;
+        }
+        output("Value of predicates is : " + predicates);
+        return predicate;
+    }
+
+    private Predicate<OperationInspectInfo> valueOrRangeInput(String conditionTpe) {
+        List<Predicate<OperationInspectInfo>> predicates = new ArrayList<>();
+        Predicate<OperationInspectInfo> predicate = null;
+        String clause = "and";
+        boolean finish = false, next = false;
+        while (!finish) {
+            if (next) {
+                clause = getStringUserInput("Select conditional operator: [and/or]");
+            }
+            final String input = getStringUserInput("Select operator : [</>/<=/>=/!=]");
+            switch (input) {
+                case LESS_THAN:
+                    long in = getLongUserInput("Enter range " + conditionTpe);
+                    predicates.add(a -> a.getSequenceNumber() < in);
+                    break;
+                case GREATER_THAN:
+                    in = getLongUserInput("Enter range " + conditionTpe);
+                    predicates.add(a -> a.getSequenceNumber() > in);
+                    break;
+                case LESS_THAN_EQUAL_TO:
+                     in = getLongUserInput("Enter range " + conditionTpe);
+                    predicates.add(a -> a.getSequenceNumber() <= in);
+                    break;
+                case GREATER_THAN_EQUAL_TO:
+                    in = getLongUserInput("Enter range " + conditionTpe);
+                    predicates.add(a -> a.getSequenceNumber() >= in);
+                    break;
+                case NOT_EQUAL_TO:
+                    in = getLongUserInput("Enter range " + conditionTpe);
+                    predicates.add(a -> a.getSequenceNumber() != in);
+                    break;
+                default:
+                    output("Invalid input please select valid operator : [</>/<=/>=/!=]");
+            }
+            predicate = clause.equals("and") ?
```

Comment:
Could this "and" be a constant as the others you have created?

Reply:
done